src/Microsoft.DotNet.Configurer/CliFolderPathCalculator.cs

```diff
@@ -18,8 +18,16 @@ public class CliFolderPathCalculator
         public string CliFallbackFolderPath => Environment.GetEnvironmentVariable("DOTNET_CLI_TEST_FALLBACKFOLDER") ??
                                                Path.Combine(new DirectoryInfo(AppContext.BaseDirectory).Parent.FullName, "NuGetFallbackFolder");
         
-        public string ExecutablePackagesPath => Path.Combine(DotnetUserProfileFolderPath, ToolsFolderName) ;
-        public readonly string ExecutablePackagesPathOnMacEnvPath = $"~/{DotnetProfileDirectoryName}/{ToolsFolderName}";
+        public string ExecutablePackagesPath => Path.Combine(DotnetUserProfileFolderPath, ToolsFolderName);
```

Comment:
I am not fond of pushing the condition for this outside of this class. Should this class make the distinction between Unix/Windows and return the appropriate ExecutablePackagesPath?

Reply:
Me too, but I cannot find a better way to model it. Help needed. Say we can return object according to OS, but what should be the interface of the return object? Once the object is returned, both Path struct on windows and unix should have the 3 method -- WithDollar,  fullpath and WithTilda which it does not fix in Windows.